baybe/searchspace/discrete.py

```diff
@@ -497,7 +498,7 @@ def drop_invalid(
             exp_rep = pd.merge(exp_rep, product_space, how="cross")
 
         # Remove entries that violate parameter constraints:
-        _apply_constraint_filter(exp_rep, constraints)
+        _apply_pandas_constraint_filter(exp_rep, constraints)
```

Comment:
you are not splitting up the constraints which results in some them being applied at least twice, once for polars once for pandas (the second time having no effect but still being unnecessary)

Reply:
I'm aware of that, but removing Polars constraints in `_apply_pandas_constraint_filter` results in `test_polars_product` to fail.

Reply:
? dont get that, the test applies the `_apply*` and product utilities separately, how is change in this function here affecting the test then?

it is simply a matter of splitting the `constraints` list here into two sublists and avoid double application of constraints

Reply:
Fixed.

Reply:
i think the split should still happen outside to avoid unnecessary double application of constraints, but thats something for the PR that does the optional application, hence closing this